### PR TITLE
🖼️  DEVEP-2684: Blog-only projects in Parliament

### PR DESCRIPTION
**Pull request by macdonst**
## Description

Support Blog only projects

## Related Issue

DEVEP-2684

## Motivation and Context

Some projects may not have doc/open api files, only blog posts so we shouldn't show the "Docs" tab in that case.

## How Has This Been Tested?

Gatsby develop/build/serve

## Screenshots (if appropriate):

Screen Shot 2021-04-27 at 15 40 41

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: gatsby-node.js [hunk 1]
```diff
@@ -685,8 +685,16 @@ exports.sourceNodes = async ({
 }) => {
   const { createNode } = actions
 
-  const data = [{ title: "Docs", path: "/" }]
-  if (glob.sync(`${process.env.LOCAL_PROJECT_DIRECTORY}/blog/**`).length > 0) {
+  const blogFiles = glob.sync(`${process.env.LOCAL_PROJECT_DIRECTORY}/blog/**`)
```
Comment (chr0n1x) [inline gatsby-node.js:688]: do we have a list of acceptable blog file extensions (e.g.: `[.md .html ...]`)? or are we really ok with all-the-things?

**Review by chr0n1x — APPROVED** [review]
looks good, just the one question

File: gatsby-node.js (hunk 1 above)
Comment (sic2) [reply gatsby-node.js:688]: This is a valid comment. I think with the current implementation you could put any file, say an image, and we'd render a site with the blog even if there isn't anything to render.

I believe Gatsby has a set of allowed extensions that make sense for a blog. However, I'd push toward having our own subset. I believe sticking with markdown for the moment is a better move. 

What do you think?

File: gatsby-node.js [hunk 2]
```diff
@@ -685,8 +685,16 @@ exports.sourceNodes = async ({
...
+    ignore: blogFiles,
+  })
+
+  const data = []
+  if (docFiles.length > 0) {
+    data.push({ title: "Docs", path: "/" })
+  }
+  if (blogFiles.length > 0) {
     data.push({ title: "Blog", path: "/blog" })
   }
```
Comment (sic2) [inline gatsby-node.js:699]: There is a pattern here. We should consider abstracting this way.
I might be wrong, but I believe in the near future we may want to support other "tracks" for the site, not just blogs and docs.

I will leave @macdonst to weigh on this.

Comment (sic2) [reply gatsby-node.js:699]: With that in mind, the setting of that `ignore: blogFiles` will become more complex over time. Something to think about.

File: gatsby-node.js (hunk 1 above)
Comment (chr0n1x) [reply gatsby-node.js:688]: yeah, IMO markdown & any supporting file types for the time being would be better (so as you said, a subset of the gatsby defaults). we could then monitor use-cases coming in from users, build up the list based on the 90% of cases.

Comment (macdonst) [reply gatsby-node.js]: Solid point folks. I've switched it to only search for markdown in the blog posts folder for now.

File: gatsby-node.js (hunk 2 above)
Comment (macdonst) [reply gatsby-node.js:700]: I think that is a great idea but a bit out of scope for this PR where we just need to roll out blog only support ASAP. You are 💯 correct we'll need to refactor this as we get into supporting other tracks.

**Pull request merged**